### PR TITLE
[Cypress] Add ui tests for lifecycle

### PR DESCRIPTION
**Pull request by Abhishek-kumar09**
Signed-off-by: Abhishek-kumar09 <abhimait1909@gmail.com>

**Description**
1. Add liefcyle (e2e) tests for active adapters

Few Problems that is hindering the further testing: 
- [x]  the navigation to some service mesh from the navigator doesn't work(https://github.com/meshery/meshery/pull/5307)
- [x] All the service mesh has another enpoint based on their name that shows the way older UI leveraging the hundreds of oam components coming from backend thus freezing the interface sometimes(https://github.com/meshery/meshery/pull/5307)
- [x] The inactive adapters UI page is the way it is expected, it should provide UI to configure which is not yet done.(https://github.com/meshery/meshery/pull/5307)

**More:**
1. Add tests for action items on the lifecycle menu
2. Add tests for unavailable adapters
3. Add tests for button clicks and deploy actions
This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Abhishek-kumar09** [comment]
The falied e2e test is the proof of the UI not working as expected.

**Comment by MarioArriaga92** [comment]
Taking a look but maybe there's a better way to find elements than using `cy.contains`? As per [Cypress Best Practices > Selecting Elements](https://docs.cypress.io/guides/references/best-practices#Selecting-Elements) use of `cy.contains('text').` is recommended on a "Depends" basis as its coupled to text content that may change. If text being asserted on is not likely to change soon or we're okay with fact that test could break if we don't update text in cy commands when we modify it, then okay to keep using it (else we should try to add 'data-cy' attribute dinamically and use it in our cypress tests, when possible)

**Comment by leecalcote** [comment]
> The falied e2e test is the proof of the UI not working as expected.

Issue(s) filed, @Abhishek-kumar09?

**Comment by warunicorn19** [comment]
@Abhishek-kumar09 a little brief on this would be great for the meshery dev call.

**Comment by leecalcote** [comment]
@MarioArriaga92 can you implement the suggestions that you've made? You can checkout the code in this PR/branch and work on it directly.

**Comment by MarioArriaga92** [comment]
Right away @leecalcote

File: ui/cypress/integration/e2e/lifecyclecheck_spec.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+/// <reference types="cypress" />
+// ***********************************************************
+
+function doGeneralIstioLifecycleChecks(cy) {
```
Comment (MarioArriaga92) [inline ui/cypress/integration/e2e/lifecyclecheck_spec.js:4]: Passing 'cy' object is not needed (available globally)

File: ui/cypress/integration/e2e/lifecyclecheck_spec.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  cy.contains("Lifecycle");
+  cy.contains("Manage Service Mesh Lifecycle");
+  cy.contains("Namespace");
+  cy.contains("Manage Sample Application Lifecycle");
+  cy.contains("Apply Service Mesh Configuration");
+  cy.contains("Validate Service Mesh Configuration");
+  cy.contains("Apply Custom Configuration");
+}
+
+function verifyAllIstioAddons(cy) {
```
Comment (MarioArriaga92) [inline ui/cypress/integration/e2e/lifecyclecheck_spec.js:14]: Passing 'cy' object is not needed (available globally)

File: ui/cypress/integration/e2e/lifecyclecheck_spec.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  cy.contains("Grafana");
+  cy.contains("Jaeger");
+  cy.contains("Kiali");
+  cy.contains("Prometheus");
+  cy.contains("Zipkin");
+}
+
+function doGeneralConsulLifecycleChecks(cy) {
```
Comment (MarioArriaga92) [inline ui/cypress/integration/e2e/lifecyclecheck_spec.js:22]: Passing 'cy' object is not needed (available globally)

File: ui/cypress/integration/e2e/lifecyclecheck_spec.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  cy.contains("Lifecycle");
+  cy.contains("Manage Service Mesh Lifecycle");
+  cy.contains("Namespace");
+  cy.contains("Manage Sample Application Lifecycle");
+  cy.contains("Apply Custom Configuration");
+  cy.get('body').should("not.contain",  "Apply Service Mesh Configuration");
```
Comment (MarioArriaga92) [inline ui/cypress/integration/e2e/lifecyclecheck_spec.js:28]: If [a test case](https://docs.google.com/spreadsheets/d/13Ir4gfaKoAX9r8qYjAFFl_U9ntke4X5ndREY1T7bnVs/edit#gid=0&range=138:138) for Consul Meshery Adapter is defined in Meshery Test Plan (“Click the ‘+’ option in the “Apply Service Mesh Configuration” card and try applying those operations that appear on the service mesh management page.“) why shouldn’t that card be present?

**Review by MarioArriaga92 — CHANGES_REQUESTED** [review]
Left a couple of comments

**Comment by leecalcote** [comment]
Syncing with `master` again.

**Comment by MarioArriaga92** [comment]
@Abhishek-kumar09 can you mention which are the open github issues that cause tests included in this PR to fail please?

**Comment by MarioArriaga92** [comment]
Btw integration test fails for filter configuration page because title element found is not "'FiltersBETA'" is this still expected? (probably related to some other PR but wanted to double-check here first)

File: ui/components/MesheryPlayComponent.js [hunk 5]
```diff
@@ -203,6 +215,7 @@ class MesheryPlayComponent extends React.Component {
     const self = this;
     const imageIcon = self.pickImage(adapter);
     let adapCount = 0;
+    console.log({ adapter, meshAdapters })
```
Comment (Abhishek-kumar09) [inline ui/components/MesheryPlayComponent.js:218]:
```suggestion
```

**Comment by Abhishek-kumar09** [comment]
@leecalcote @MarioArriaga92 Can we move here?

**Review by leecalcote — APPROVED** [review]
👨‍🎤 and roll.

**Pull request merged**